src/org/traccar/protocol/WatchFrameDecoder.java

```diff
@@ -30,8 +30,34 @@
     protected Object decode(
             ChannelHandlerContext ctx, Channel channel, ByteBuf buf) throws Exception {
 
+        int idFollowUpFrameStart;
```

Comment:
this part removes bytes from the stream which are not placed between a frame start and frame end marker. Also the the usecase of partial frames ( another frame start marker without an frame end marker) is handled here